docs/sources/tempo/release-notes/v2-4.md

```diff
@@ -0,0 +1,248 @@
+
+### TraceQL metrics (experimental)
+
+We’re excited to announce the addition of metrics queries to the TraceQL language. Metric queries extend trace queries by applying a function to trace query results. This powerful feature creates metrics from traces, much in the same way that LogQL metric queries create metrics from logs.
+
+Tempo 2.4 introduces metrics queries to the TraceQL language as an experimental feature.
+
+Metric queries extend trace queries by applying a function to trace query results.
+
+This powerful feature creates metrics from traces, much in the same way that LogQL metric queries create metrics from logs.
+
+Initially, only `count_over_time` and `rate` are supported.
+
+For example:
+
+```
+{ resource.service.name = "foo" && status = error } | rate()
+```
+
+In this case, we are calculating the rate of the erroring spans coming from the service `foo`. Rate is a `spans/sec` quantity.
+
+```
+{ resource.service.name = "foo" && status = error } | rate() by (span.http.route)
+```
+
+Now, we are still rating the erroring spans in the service `foo` but the metrics have been broken
+down by HTTP endpoint. This might let you determine that `/api/sad` had a higher rate of erroring
+spans than `/api/happy`, for example.
+
+After you have configured your Tempo data source in Grafana or Grafana Cloud, you can use Grafana Explore to [query and visualize the metrics]({{< relref "../operations/traceql-metrics" >}}).
+
+![Metrics visualization in Grafana](/media/docs/tempo/metrics-explore-sample-2.4.png)
+
+For more information, refer to the [TraceQL metrics]({{< relref "../operations/traceql-metrics" >}}) documentation. [PRs [3227](https://github.com/grafana/tempo/pull/3227) [#3252](https://github.com/grafana/tempo/pull/3252), [3258](https://github.com/grafana/tempo/pull/3258)]
+
+To learn more about the TraceQL syntax, see the [TraceQL documentation]({{< relref "../traceql" >}}). For information on planned future extensions to the TraceQL language, refer to [future work]({{< relref "../traceql/architecture" >}}).
+
+#### TraceQL performance improvements
+
+In addition, we’ve improved performance for queries and structural operators
+
+* Improve TraceQL regex performance in certain queries. [PR  [3139](https://github.com/grafana/tempo/pull/3139)]
+* Improve TraceQL performance in complex queries. [[PR 3113](https://github.com/grafana/tempo/pull/3113)]
+* TraceQL/Structural operators performance improvement. [[PR 3088](https://github.com/grafana/tempo/pull/3088)]
+
+### vParquet3 is now the default block format
+
+Tempo 2.4 makes [vParquet3]({{< relref "../configuration/parquet" >}}) default, a Parquet version designed to be more compatible with other Parquet implementations, available as a production-ready option. This block format improves query performance relative to previous formats. [[PR 2526](https://github.com/grafana/tempo/pull/2526)]
```

Comment:
> a Parquet version designed to be more compatible with other Parquet implementations

i don't think this is true. we should just double down on the addition of dedicated columns which was the real reason for vp3. should we link to the blogpost as well as the docs?

Reply:
We have a link to the blog post earlier in the release notes, but you're right that it should be here too. 

```suggestion
Tempo 2.4 makes [vParquet3]({{< relref "../configuration/parquet" >}}) default.
Dedicated attribute columns, available using vParquet3, improve query performance by storing the largest and most frequently used attributes in their own columns, rather than in the generic attribute key-value list.
For more information, refer to the 
[Accelerate TraceQL queries at scale with dedicated attribute columns in Grafana Tempo](/blog/2024/01/22/accelerate-traceql-queries-at-scale-with-dedicated-attribute-columns-in-grafana-tempo/) blog post. [[PR 2526](https://github.com/grafana/tempo/pull/2526)]
```

Reply:
not using my suggestino in favor of Jen's.